include/caf/caf_overview.rst

```diff
@@ -0,0 +1,182 @@
+.. _caf_overview:
+
+Common Application Framework overview
+#####################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+Common Application Framework (CAF) is a set of ready-for-use modules and events built on top of :ref:`event_manager`.
+Using CAF allows you to have a consistent event-based architecture in your application.
+
+One of the applications that are using the CAF modules in |NCS| is :ref:`nrf_desktop`.
```

Comment:
I liked it better few lines below. This could be put in `note`. Right now its in the middle of CAF overview description.

Reply:
Added a new section after this line.